manager/controlapi/service.go

```diff
@@ -433,7 +433,7 @@ func (s *Server) validateNetworks(networks []*api.NetworkAttachmentConfig) error
 		if network == nil {
 			continue
 		}
-		if network.Spec.Internal {
+		if network.Spec.Ingress {
 			return grpc.Errorf(codes.InvalidArgument,
 				"Service cannot be explicitly attached to %q network which is a swarm internal network",
```

Comment:
Makes sense, now that the ingress network can be created with a different name, it is better to fix the message.